hypervisor/lib/crypto/mbedtls/md.c

```diff
@@ -31,191 +31,158 @@
 /*
  * Reminder: update profiles in x509_crt.c when adding a new hash!
  */
-static const int32_t supported_digests[] = {
+static const mbedtls_md_type_t supported_digests[] = {
         MBEDTLS_MD_SHA256,
         MBEDTLS_MD_NONE
 };
 
-const int32_t *mbedtls_md_list( void )
+const mbedtls_md_type_t *mbedtls_md_list( void )
 {
     return( supported_digests );
 }
 
 const mbedtls_md_info_t *mbedtls_md_info_from_type( mbedtls_md_type_t md_type )
 {
-    switch( md_type )
+    const mbedtls_md_info_t *md_info = NULL;
+
+    switch ( md_type )
     {
         case MBEDTLS_MD_SHA256:
-            return( &mbedtls_sha256_info );
+            md_info = &mbedtls_sha256_info;
+            break;
         default:
-            return( NULL );
+            break;
     }
+
+    return md_info;
 }
 
 void mbedtls_md_init( mbedtls_md_context_t *ctx )
 {
-    memset( ctx, 0, sizeof( mbedtls_md_context_t ) );
+    (void) memset( ctx, 0U, sizeof( mbedtls_md_context_t ) );
 }
 
 void mbedtls_md_free( mbedtls_md_context_t *ctx )
 {
-    if( ctx == NULL )
-        return;
-
-    mbedtls_platform_zeroize( ctx, sizeof( mbedtls_md_context_t ) );
-}
-
-int32_t mbedtls_md_clone( mbedtls_md_context_t *dst,
-                      const mbedtls_md_context_t *src )
-{
-    if( dst == NULL || dst->md_info == NULL ||
-        src == NULL || src->md_info == NULL ||
-        dst->md_info != src->md_info )
-    {
-        return( MBEDTLS_ERR_MD_BAD_INPUT_DATA );
+    if ( ctx != NULL ) {
+        (void) mbedtls_platform_zeroize( ctx, sizeof( mbedtls_md_context_t ) );
```

Comment:
Please remove the unnecessary space. Same for the others.
`(void)mbedtls_platform_zeroize(ctx, sizeof(mbedtls_md_context_t));`